site/en/docs/extensions/reference/declarativeNetRequest/index.md

```diff
@@ -221,9 +221,24 @@ manifest field to limit which Chrome versions can install your extension.
 
 The number of rules available after that depends on how many rules are enabled by all the extensions installed on a user's browser. You can find this number at runtime by calling [`getAvailableStaticRuleCount()`](#method-getAvailableStaticRuleCount). You can see [an example of this](#update-static-rulesets) under [code examples](#code-examples).
 
-#### Dynamic and session rules {: #dynamic-session-rules }
+#### Session rules {: #session-rules }
 
-The limits applied to dynamic and session rules are simpler than static rules. The total number of both cannot exceed 5000. This is called the [`MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES`](#property-MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES).
+An extension can have up to 5000 session rules. This is exposed as the
+[`MAX_NUMBER_OF_SESSION_RULES`](#property-MAX_NUMBER_OF_SESSION_RULES).
+
+Before Chrome 120, there was a limit of 5000 combined dynamic and session rules.
+
+#### Dynamic rules {: #dynamic-rules }
+
+An extension can have at least 5000 dynamic rules. This is exposed as the
```

Comment:
The flow of this section is a bit hard to follow. I am writing the facts in bullets to check if I'm understanding correctly:

**Dynamic rule types:**
- **Safe Rules**: those with actions of block, allow, allowAllRequests, or upgradeScheme.
- **Unsafe Rules**: any rules with actions not included under the safe rules.

**Chrome 121 and Later:**
- A higher limit of 30,000 for safe dynamic rules, indicated as `MAX_NUMBER_OF_DYNAMIC_RULES`.
- The limit of 5,000 unsafe dynamic rules, denoted as `MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES`, also contributes towards the 30,000 safe rule limit.

**Before Chrome 121:**
A combined limit of 5,000 for both dynamic and session rules, without a specific distinction between safe and unsafe rules.

Reply:
This looks good to me. Glad you could follow at least, but happy to iterate. Agree this is a lot.

Reply:
Let's leave it as is for the MVP.